metisp/pymetis/src/pymetis/classes/products/common.py

```diff
@@ -50,6 +53,41 @@ class BandSpecificProduct(PipelineProduct):
     def band(cls) -> str:
         """Return the internal band, or a placeholder for manpage."""
         return cls._band or "{band}"
+    
+class TableProduct(PipelineProduct):
+    """Product with table data"""
+
+    def __init__(self,
+                    recipe_impl: 'MetisRecipeImpl',
+                    header: cpl.core.PropertyList,
+                    table: cpl.core.Table):
+        super().__init__(recipe_impl, header, image=None)
+        self.table: cpl.core.Table = table
+
+    def save(self):
+        """ Save this Product to a file """
+        Msg.info(self.__class__.__qualname__,
+                 f"Saving product file as {self.output_file_name!r}:")
+        Msg.info(self.__class__.__qualname__,
+                 f"All frames ({len(self.recipe.frameset)}): {sorted([frame.tag for frame in self.recipe.frameset])}")
+        Msg.info(self.__class__.__qualname__,
+                 f"Loaded frames ({len(self.recipe.valid_frames)}): {sorted([frame.tag for frame in self.recipe.valid_frames])}")
+        # Check that the tag matches the generic regex
+        assert self._regex_tag.match(self.tag()) is not None, \
+            f"Invalid {self.__class__.__qualname__} product tag '{self.tag()}'"
+        # At least one frame in the recipe frameset must be tagged as RAW!
+        # Otherwise, it *will not* save (rite of passage)
+        cpl.dfs.save_table(
+            self.recipe.frameset,       # All frames for the recipe
+            self.recipe.parameters,     # The list of input parameters
+            self.recipe.valid_frames,   # The list of frames actually used FixMe currently not working as intended
+            self.table,                 # Table to be saved
+            self.recipe.name,           # Name of the recipe
+            self.properties,            # Properties to be appended
+            PIPELINE,
+            self.output_file_name,
+            header=self.header,
```

Comment:
Just to make sure we're not repeating the previous error, when QC parameters were in self.header instead of self.properties and therefore not saved. But the class parameter `header` is given to `self.properties` by the parent class, isn't it? And here, you are simply saving the product header to `self.header`, just in case?